Assignment/SampleData.cs

```diff
@@ -0,0 +1,104 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Assignment.Interfaces;
+
+namespace Assignment;
+
+public class SampleData : ISampleData
+{
+    // 1.
+    public IEnumerable<string> CsvRows
+        {
+            get
+            {
+                // Specify the path to the People.csv file
+                string csvFilePath = "People.csv";
+
+                // Check if the file exists
+                if (File.Exists(csvFilePath))
+                {
+                    // Read all lines from the file and return as enumerable, skipping the first row
+                    return File.ReadLines(csvFilePath).Skip(1);
+                }
+                else
+                {
+                    // If the file doesn't exist, throw an exception or handle it appropriately
+                    throw new FileNotFoundException("People.csv file not found.");
+                }
+            }
+        }
+    // 2.
+        public IEnumerable<string> GetUniqueSortedListOfStatesGivenCsvRows()
+        {
+            // Read CSV rows from CsvRows property
+            var csvRows = CsvRows;
+
+            // Extract states and remove duplicates
+            var stateSet = new HashSet<string>();
+            foreach (var row in csvRows)
+            {
+                var columns = row.Split(',');
+                if (columns.Length >= 7) // Assuming state is at index 6
+                {
+                    stateSet.Add(columns[6].Trim()); // Trim to remove any extra spaces
+                }
+            }
+

```

Comment:
I'm not sure if this is using linq to accomplish the specification? this is what we did:

```suggestion
return CsvRows.Select(item => item.Split(',')[6])
       .OrderBy(state => state)
       .Distinct();//for unique

```